src/Microsoft.AspNetCore.Mvc.Razor.Extensions/RazorExtensions.cs

```diff
@@ -29,10 +29,13 @@ public static void Register(IRazorEngineBuilder builder)
             builder.Features.Add(new ViewComponentTagHelperPass());
             builder.Features.Add(new RazorPageDocumentClassifierPass());
             builder.Features.Add(new MvcViewDocumentClassifierPass());
-            builder.Features.Add(new AssemblyAttributeInjectionPass());
 
             if (!builder.DesignTime)
             {
+                // This is a temporary workaround until we have support for multi-targeting. If this were to be added
+                // then in 1.0.0 applications you'd get an error about missing attributes at design time.
+                builder.Features.Add(new AssemblyAttributeInjectionPass());
```

Comment:
I actually think this is fine as a non-workaround. We don't need this in the design time code.